ios/IoReactNativeIntegrity.swift

```diff
@@ -1,8 +1,187 @@
+    
+    guard #available(iOS 14.0, *) else {
+      ME.unsupportedIOSVersion.reject(reject: reject)
+      return
+    }
+    
+    if DCAppAttestService.shared.isSupported {
+      let service = DCAppAttestService.shared
+      
+      // Safely encode the clientData string to data and generate a hash.
+      guard let encodedClientData = clientData.data(using: .utf8) else {
+        ME.clientDataEncodingError.reject(reject: reject)
+        return
+      }
+      
+      let clientDataHash = Data(SHA256.hash(data: encodedClientData))
+      
+      service.generateAssertion(hardwareKeyTag, clientDataHash: clientDataHash) { assertion, error in
+        
+        guard error == nil else {
+          ME.generationAssertionFailed.reject(reject: reject, ("error", error?.localizedDescription ?? ""))
+          return
+        }
+        
+        let encodedAssertion = assertion?.base64EncodedString()
+        resolve(encodedAssertion)
+        
+      }
+    } else {
+      ME.unsupportedService.reject(reject: reject)
+    }
   }
+  
+  /// A private enum to handle exceptions and errors.
+  /// - generationKeyFailed: The hardware key generation failed.
+  /// - unsupportedService: The DeviceCheck App Attestation Service is not supported.
+  /// - attestationError: The attestation request failed.
+  /// - unsupportedIOSVersion: The iOS version is not supported.
+  /// - challengeError: The challenge encoding failed.
+  /// - clientDataEncodingError: The clientData encoding failed.
+  /// - generationAssertionFailed: The assertion generation failed.
+  private enum ModuleException: String, CaseIterable {
+    
+    case generationKeyFailed = "GENERATION_KEY_FAILED"
+    case unsupportedService = "UNSUPPORTED_SERVICE"
+    case attestationError = "ATTESTATION_ERROR"
+    case unsupportedIOSVersion = "UNSUPPORTED_IOS_VERSION"
+    case challengeError = "CHALLANGE_ERROR"
+    case clientDataEncodingError = "CLIENT_DATA_ENCODING_ERROR"
+    case generationAssertionFailed = "GENERATION_ASSERTION_FAILED"
+      
+      func error(userInfo: [String : Any]? = nil) -> NSError {
+        switch self {
+        case .generationKeyFailed:
+          return NSError(domain: self.rawValue, code: -1, userInfo: userInfo)
+        case .unsupportedService:
+          return NSError(domain: self.rawValue, code: -1, userInfo: userInfo)
+        case .attestationError:
+          return NSError(domain: self.rawValue, code: -1, userInfo: userInfo)
+        case .unsupportedIOSVersion:
+          return NSError(domain: self.rawValue, code: -1, userInfo: userInfo)
+        case .challengeError:
+          return NSError(domain: self.rawValue, code: -1, userInfo: userInfo)
+        case .clientDataEncodingError:
+          return NSError(domain: self.rawValue, code: -1, userInfo: userInfo)
+        case .generationAssertionFailed:
+          return NSError(domain: self.rawValue, code: -1, userInfo: userInfo)
+        }
```

Comment:
I alway suggest to run the code formatter on files before committing. On Xcode the shortcut it `Ctrl+I`.